esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -713,47 +713,47 @@ bool CFileSprayEx::onDFUWUSearch(IEspContext &context, IEspDFUWUSearchRequest &
     return true;
 }
 
-int readFromCommaSeparatedString(const char *commaSeparatedString, StringBuffer* output)
+//Sasha List command returns comma separated strings about archived DFU WUs.
+//Each comma separated string is for one DFU WU: WUID,User,JobName,ClusterName,StateMessage,Command.
+//Parse the string to build one DFU WU.
+IEspDFUWorkunit *CFileSprayEx::createDFUWUFromSashaListResult(const char *result)
 {
-    int numOfItems = 0;
-    if (commaSeparatedString && *commaSeparatedString)
-    {
-        char *pStr = (char *) commaSeparatedString;
-        while (pStr)
-        {
-            char item[1024];
-            bool bFoundComma = false;
-            int len = strlen(pStr);
-            for (int i = 0; i < len; i++)
-            {
-                char *pStr1 = pStr + i;
-                if (pStr1[0] != ',')
-                    continue;
-
-                strncpy(item, pStr, pStr1 - pStr);
-                item[pStr1 - pStr] = 0;
+    StringArray wuInfoFields;
+    wuInfoFields.appendList(result, ",");
+    unsigned numOfWUInfoFields = wuInfoFields.length();
+    if (numOfWUInfoFields == 0)
+        return nullptr;
 
-                bFoundComma = true;
-                if (i < len - 1)
-                    pStr = pStr1 + 1;
-                else
-                    pStr = NULL;
+    Owned<IEspDFUWorkunit> dfuWU = createDFUWorkunit();
+    dfuWU->setArchived(true);
 
+    ForEachItemIn(i, wuInfoFields)
+    {
+        switch(i)
+        {
+            case 0:
+                dfuWU->setID(wuInfoFields.item(i));
+                break;
+            case 1:
+                dfuWU->setUser(wuInfoFields.item(i));
+                break;
+            case 2:
+                dfuWU->setJobName(wuInfoFields.item(i));
+                break;
+            case 3:
+                dfuWU->setClusterName(wuInfoFields.item(i));
+                break;
+            case 4:
+                dfuWU->setStateMessage(wuInfoFields.item(i));
+                break;
+            case 5:
+                const char *commandStr = wuInfoFields.item(5);
+                if (!isEmptyString(commandStr))
+                    dfuWU->setCommand(decodeDFUcommand(commandStr));
```

Comment:
This is a change from the previous behaviour.  Does it need documenting?  Will it have any backward compatibility implications?

Reply:
@ghalliday The existing code does not work (a bug for a long time). The code expects the commandStr to be an ID (number) representing a DFU command. For example, '1' for copy, '6' for import, etc. But, the commandStr is a string, ex. 'copy', 'import', etc. Should it be documented as a bug fix? For backward compatibility, should I add a version (keep the bug as it is for previous version)?

Reply:
If the argument is a number you could use that, otherwise convert the dfu command.  Of convert the dfu command, and if it is DFUcmd_none fall back to the old behaviour.

Reply:
I will do.

Reply:
I wasn't clear - that was two different approaches.  I think the second one is simpler code so I would stick with that.